src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+
+using System;
+using System.Net.Http;
+using System.Net;
+
+namespace System.Net.Http
+{
+    sealed internal class HttpEnvironmentProxy : IWebProxy {
+        private Uri _http = null;    // String URI for HTTP requests
+        private Uri _https = null;   // String URI for HTTPS requests
+        private NetworkCredential _httpCred;
+        private NetworkCredential _httpsCred;
+        String[] _bypass = null;    // list of domains not to proxy
+
+
+        public static HttpEnvironmentProxy TryToCreate()
+        {
+            string http = null;     // local variable to hold most specific value
+            string https = null;
+
+
+            // get environmental variables. protocol specific take precedence over
+            // general all_*, lover case variable has precedence over upper case.
+            // note that curl uses HTTPS_PROXY but not HTTP_PROXY.
+            // For http, only http_proxy and generic variables are used.
+            string value = Environment.GetEnvironmentVariable("ALL_PROXY");
+            if (value != null)
```

Comment:
Should this be made as find the most preferred first, and if value is null, read other env variables, like

```c#
if (most_preferred)
{
}
else if (next_preferred)
{
}
.. so on
```

Anyways the value is being overridden in http, https, so if most preferred is found, we don't have to read other least preferred env variables.

Reply:
In that case you will read all variables if only all_proxy is set.
I'd be happy to change that but I feel the cost will be similar without data what users us.

Reply:
The cost is not similar, as it is now, you will be reading the environment variables all the time, as supposed to maybe reading once if the most preferred is set, or reading all is only least preferred is set.

Reply:
I agree with @Priya91.  As it currently stands, we're having to check all of the environment variables, whereas if the most specific were checked first, we'd only need to check until we find an answer or until we exhaust them, so it should be as good or better.